dexter-jenkins/dexter-jenkins/src/main/java/io/jenkins/plugins/dexter/DexterPublisher.java

```diff
@@ -238,27 +226,38 @@ public boolean perform(AbstractBuild build, Launcher launcher, BuildListener lis
 				} else
 					message = "Error saving logs";
 			} else {
+
 				writeToShFile();
-				Runtime runtime = Runtime.getRuntime();
-				String command = "/home/v.sanko/dexter-server/run.sh";
-				try {
-					runtime.exec(command);
-				} catch (IOException e) {
-					e.printStackTrace();
-				}
 				writeToFile();
 
-				String commandDexter = pathToBat + "/dexter.sh " + dexterUser + " " + dexterPassword + " > logs.txt \"";
 				try {
-					runtime.exec(commandDexter);
-				} catch (IOException e) {
+
+					int ch;
+
+					ProcessBuilder pb = new ProcessBuilder(dexterServerPath + "/run.sh");
+					pb.directory(new File(dexterServerPath));
+
+					Process shellProcess = pb.start();
+
+					ProcessBuilder pbClient = new ProcessBuilder(pathToBat + "/dexter.sh", dexterUser, dexterPassword);
+					pbClient.directory(new File(pathToBat));
+
+					Process shellProcessClient = pbClient.start();
+
+					InputStreamReader myIStreamReader = new InputStreamReader(shellProcessClient.getInputStream());
+
+					while ((ch = myIStreamReader.read()) != -1) {
```

Comment:
This seem pretty-low level. Cannot it be read a whole String at once? e.g. using stream API:
`String result = new BufferedReader(myIStreamReader).lines().collect(Collectors.joining(System.lineSeparator()));`

Also, am I correct that it destroys the process (`shellProcess.destroy();`) after reading each character?

Reply:
While jenkins plugin are builded they are checked with findbugs. We can use your method but findbugs will return an error cause java.util.stream.Collectors doesn't have default encoding (even though  it will be working solution) build will be failed. So I decided to use this low level solution to avoid findbugs error. But I can change that and it will require just disable findbugs while building project.

Reply:
And you are right - process should not be destroyed every time after reading, change that and now it is destroing after reading all symbols